jaeger_client/metrics.py

```diff
@@ -21,69 +21,122 @@
+        :param tags: tags for the timer
+        :return: a callable function which takes the duration to
+        record ie. def record(duration)
+        """
+        return self._noop
 
-    TRACES_JOINED_NOT_SAMPLED = '%s.traces-joined.not-sampled' % prefix
-    """Number of externally started non-sampled traces this tracer joined"""
+    def create_gauge(self, name, tags=None):
+        """
+        Generates a new gauge from the given name and tags and returns
+        a callable function used to update the gauge.
+        :param name: name of the gauge
+        :param tags: tags for the gauge
+        :return: a callable function which takes the value to update
+        the gauge with ie. def update(value)
+        """
+        return self._noop
 
-    SPANS_SAMPLED = '%s.spans.sampled' % prefix
-    """Number of sampled spans started/finished by this tracer"""
+    def _merge_tags(self, tags=None):
+        if not self._tags:
+            return tags
+        tags_cpy = self._tags.copy()
+        tags_cpy.update(tags)
+        return tags_cpy
 
-    SPANS_NOT_SAMPLED = '%s.spans.not-sampled' % prefix
-    """Number of not-sampled spans started/finished by this tracer"""
 
-    TRACER_DECODING_ERRORS = '%s.decoding.errors' % prefix
-    """Number of errors decoding tracing context"""
+class LegacyMetricsFactory(MetricsFactory):
+    """A MetricsFactory wrapper around Metrics."""
 
-    REPORTER_SUCCESS = '%s.spans.reported' % prefix
-    """Number of spans successfully reported"""
+    def __init__(self, metrics, tags=None):
+        super(LegacyMetricsFactory, self).__init__(
+            tags=tags
+        )
+        self._metrics = metrics
 
-    REPORTER_FAILURE = '%s.spans.failed' % prefix
-    """Number of spans in failed attempts to report"""
+    def create_counter(self, name, tags=None):
+        key = self._get_key(name, self._merge_tags(tags))
 
-    REPORTER_DROPPED = '%s.spans.dropped' % prefix
-    """Number of spans dropped from the reporter"""
+        def increment(value):
+            return self._metrics.count(key, value)
+        return increment
 
-    REPORTER_SOCKET = '%s.spans.socket_error' % prefix
-    """Number of spans dropped due to socket error"""
+    def create_timer(self, name, tags=None):
+        key = self._get_key(name, self._merge_tags(tags))
 
-    REPORTER_QUEUE_LENGTH = '%s.queue_length' % prefix
-    """Current number of spans in the reporter queue"""
+        def record(value):
+            return self._metrics.timing(key, value)
+        return record
 
-    SAMPLER_ERRORS = '%s.sampler.errors' % prefix
-    """Number of times the Sampler failed to retrieve sampling strategy"""
+    def create_gauge(self, name, tags=None):
+        key = self._get_key(name, self._merge_tags(tags))
 
-    def __init__(self, count=None, gauge=None):
+        def update(value):
+            return self._metrics.gauge(key, value)
+        return update
+
+    def _get_key(self, name, tags=None):
```

Comment:
_merge_tags() will be used by both LegacyMetricsFactory and M3MetricsFactory so I put it in the base class and added tags as a parameter to the base class. I guess it's all possible to make _merge_tags static but I think this is cleaner. Also, _get_key will only be used by LegacyMetricsFactory so think it's fine as is.